### ISSUE TITLE
Custom logging formats

### ISSUE THREAD

**Issue by domtronn** [bbc/flashheart#64]
It would be nice to change the logging format of **Flashheart**.

For data/log visualisers that leverage JSON logs like [**Kibana**](https://www.elastic.co/products/kibana), having the upstream access log as a single message becomes hard to parse, _i.e._

```sh
GET http://echo.jsontest.com/hello/world 200 10 ms
```

Libraries like [**Morgan**](https://github.com/expressjs/morgan#using-format-string-of-predefined-tokens) allow for custom format functions, _e.g._

```js
morgan((tokens, req, res) => tokens)
```

Which would return the JSON object of all of the tokens that **Morgan** monitors. 

Since **Flashheart** allows you to inject your own _logger_, it makes sense _(to me at least)_ to be able to define your own _formatter_ to return the right thing for the logger to consume.

This could also give users access to _more_ information that's on the request and response options, I would like to be able to define something like

```js
const format = (req, res) => ({ short: `${req.method} - ${res.statusCode}`, time: res.elapsedTime })

createClient({ format, logger })
```

And this could easily be done [here](https://github.com/bbc/flashheart/blob/master/lib/client.js#L108-L116) with something like;

```js
Client.prototype._log = function (res) {
  var debugMessage = util.format(res.request.method, res.request.href, res.statusCode, res.elapsedTime, 'ms');

  debug(debugMessage);

  if (this.logger) {
    var logMessage = this.format 
      ? this.format(res,request, res) 
      : debugMessage

    this.logger.info(logMessage);
  }
};
```

Assuming that you still want the debug message to be string only.

What do you guys think?

**Comment by nspragg** [comment]
Thanks for raising this. Being able to format the message seems like a good idea. However, we're not looking to extend the behaviour of the current version (important patches only). We're currently writing a new versions, which should allow you make customise log message easily. 

I will keep you updated, when the new version is ready.

**Comment by domtronn** [comment]
@nspragg Have you got any estimates of when this _"new version"_ will be available? Is there any way of contributing to it or monitoring it's development?
